src/events.ts

```diff
@@ -1,41 +1,50 @@
-enum messages {
+enum partnerEvents {
   // It's really unfortunate that we call this event "accountProvisioned",
   // when what we mean is "accountProvisionInitiated". But that's the current
   // name that is being used by the dreams backend
-  accountProvisioned = 'accountProvisioned',
+  accountProvisionInitiated = 'accountProvisioned',
+  investmentAccountProvisionInitiated = 'investmentAccountProvisionInitiated',
   updateToken = 'updateToken',
   navigateTo = 'navigateTo'
 }
 
 type Message = {
-  message: {
-    requestId: string;
-    idToken?: string;
-  };
+  requestId: string;
 }
 
-type ShareMessage = {
-  message: {
-    url: string;
-    text: string;
-    title: string;
-  };
+type ShareMessage = Message & {
+  url: string;
+  text: string;
+  title: string;
 }
 
-type MessageEvent = Message & {
-  event: messages;
+type TokenDidExpireMessage = Message & {
+  idToken: string;
 }
 
-type IdTokenDidExpireEvent =  Message & {
+type InvestmentAccountProvisionMessage = Message & {
+  accountId: string;
+  dreamId: string;
```

Comment:
Please remind me, is `dreamId` necessary to expose?

Reply:
well, how we will connect account with the dream once the provisioning is done? I think we have to expose it

Reply:
Ah. The partner needs to call `POST /customers/<customer_id>/dreams/<dreamId>/account-provisioned` ?
But in that case, why are we sending the `accountId`?

I'm thinking we could store a tentative `accountId` on the dream, and then have an endpoint like:

```
POST /customers/<customer_id>/investment-accounts/accountId/provisioned
```

And then we dig up the dream with the corresponding `accountId`.
Or add an optional parameter to the old endpoint:

```
POST /customers/<customer_id>/account-provisioned
{
  accountId: string // defaults to the user savings account.
}
```

I think storing the accountId on the dream can have other benefits as well. E.g. we can always send the same id to the partner to make account provisioning idempotent.
WDYT?

Reply:
That does sound sound. I'll check in DES how the endpoint works and make changes

Reply:
Removed `dreamId` let me know if that looks okay now

Reply:
Perfect!